sdk/inc/azure/core/az_credentials.h

```diff
@@ -92,53 +50,6 @@ typedef struct
   } _internal;
 } _az_credential;
```

Comment:
Now this file only contains internal-only types and things. Move it to an internal header (`internal/az_credentials_internal.h`) or rename it to `_az_credentials.h` since the customer shouldn't be including it for anything.

Reply:
Do we still need `_az_credential_set_scopes_fn`?

Reply:
`AZ_CREDENTIAL_ANONYMOUS` is exactly what the users are going to be using, and they should include `az_credentials.h` for that. Later, as/if we implement more credential types (resurrect Client Secret Credential), they will go into this file. So I think it is useful to keep this file structure.

Reply:
At least for where we have landed now, it is certainly strange for the user to include an entirely separate header file just for one single line macro.

@barcharcraz - what do you think?

Reply:
Also, @JeffreyRichter - thoughts on keeping `az_credential.h` for `AZ_CREDENTIAL_ANONYMOUS`?

Reply:
Can we remove it for now?